presto-docs/src/main/sphinx/connector/hive-caching.rst

```diff
@@ -177,16 +177,17 @@ The implementation of the cache exposes a `number of metrics
 :doc:`inspect these and other metrics directly in Presto with the JMX connector
 or in external tools </admin/jmx>`.
 
-The following example query returns the total amount of data fetched by the
-BookKeeper on the coordinator and all the separate nodes on the workers in the
-default async mode:
+If you are using JMX connector, basic caching statistics are exposed via
+``jmx.current."rubix:catalog=<catalog_name>,name=stats"`` table.
+You can read more detailed statistics by querying ``jmx.current."rubix:catalog=<catalog_name>,name=stats``.
+You need to replace ``<catalog_name>`` with name of you Hive catalog.
+
+The following example query returns the total amount of data fetched from source
+filesystem by all Presto nodes:
 
 .. code-block:: sql
 
-  SELECT fs.readfromremote + async_downloaded_mb.count
-  FROM jmx.current."rubix:catalog=hive,name=stats" fs,
-       jmx.current."metrics:name=rubix.bookkeeper.count.async_downloaded_mb" async_downloaded_mb
-  WHERE fs.node = async_downloaded_mb.node;
+  SELECT sum(mb_read_from_source) FROM jmx.current."rubix:catalog=hive,name=stats";
```

Comment:
this is only valid for read-though mode, for async mode, we still need to account for `jmx.current."metrics:name=rubix.bookkeeper.count.async_downloaded_mb" `

https://github.com/qubole/rubix/pull/438 will introduce `total_system_source_mb_read` global metrics

Reply:
Thanks. I chaned the example query to return average cache hit. I would rather not mix-in `metrics:name=rubix.bookkeeper.count.async_downloaded_mb` into example as it does not play well with presto (no catalog name in key).

Reply:
> I would rather not mix-in metrics:name=rubix.bookkeeper.count.async_downloaded_mb into example as it does not play well with presto (no catalog name in key).

Right, `cache_hit` is fine, yet it doesn't tell the whole store. How do you know if Presto is not going directly to data source?

Reply:
We do not. But I think the bottom line is that we just cannot reasonably provide this information until qubole/rubix#438.
Well we can as long as only single connector uses caching, but I do not want to elaborate too much about that in the documentation.

I would rather push qubole/rubix#438 and update the documentation with example regarding "data read from source" as it gets merged and we update rubix.

Reply:
That sounds reasonable to me.